### PR TITLE
Update fontSize to typography

### PR DESCRIPTION
**Pull request by MarieComet**
Documentation for fontSize is not up to date, since it has to be enable in typography then fontSize supports.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: docs/reference-guides/block-api/block-supports.md [hunk 1]
```diff
@@ -424,31 +424,40 @@ supports: {
 }
 ```
 
-## fontSize
+## typography
```
Comment (youknowriad) [inline docs/reference-guides/block-api/block-supports.md:427]: I actually think it's better to leave "fontSize" (and description) here but just update the code to include the typography key.

Any thoughts here cc @oandregal @JustinyAhin

**Review by youknowriad — COMMENTED** [review]
Thanks for the fix :) Good catch. It also makes me think that now we have a lot more block supports to document. 

I guess ideally we should add them here as we stabilize them.

cc @aaronrobertshaw @andrewserong as you both work on block supports a lot these days.

**Comment by aaronrobertshaw** [comment]
Thanks for putting this PR together @MarieComet 👍 

I think the best option for the docs is a third option to those already presented. Rather than switch `fontSize` to `typography` or leave it and just reflect its nesting under `typography` in the code snippets, we should make it consistent with how the `color` supports are documented.

By this I mean, we document `typography` as we do `color`. Then also document `typography.fontSize` and `typography.lineHeight`.

I didn't want to be presumptuous and push changes to this PR but given my local patch contained markdown I couldn't work out how to include that in this comment. Instead, I created an [alternate PR](https://github.com/WordPress/gutenberg/pull/36894) containing my proposed updates. Please feel free to include those within this PR if we decide to go in that direction and I can close the alternate PR later.

>  It also makes me think that now we have a lot more block supports to document.
> 
> I guess ideally we should add them here as we stabilize them.

Agreed. As the currently experimental features get stabilized, the plan is to document them here.

**Comment by MarieComet** [comment]
That's sound good to me to close my PR in favor of yours

**Comment by MarieComet** [comment]
Closed in favor of https://github.com/WordPress/gutenberg/pull/36894

File: docs/reference-guides/block-api/block-supports.md (hunk 1 above)
Comment (oandregal) [reply docs/reference-guides/block-api/block-supports.md:427]: Hey, I do think we need to do with `typography` what we've done for `color`: have typography as a heading 2 and then the sub-keys as heading 3. Same for border, spacing, layout.

A good first step could be to create a new typography section that contains the existing fontSize and lineHeight. Then, add the undocumented ones. Note, however, that `supports.typography: true` does nothing, the properties need to be enabled individually ([code](https://github.com/WordPress/gutenberg/blob/8d258e505ceb35ac256fcf2c52d973dff286a2cf/packages/block-editor/src/hooks/typography.js#L235)).

Comment (oandregal) [reply docs/reference-guides/block-api/block-supports.md:427]: @mkaz and @aaronrobertshaw as per a conversation we've had about documenting block supports that are still experimental or in the process of becoming matured APIs.

Comment (oandregal) [reply docs/reference-guides/block-api/block-supports.md:427]: Oh, brilliant. Hadn't seen Aaron PR. That's the direction we want :)

**Pull request closed without merging**